kiwixbuild/platforms/wasm.py

```diff
@@ -0,0 +1,100 @@
+from .base import PlatformInfo
+
+from kiwixbuild.utils import pj
+from kiwixbuild._global import get_target_step
+
+
+class WasmPlatformInfo(PlatformInfo):
+    name = 'wasm'
+    static = True
+    build = 'wasm'
+    arch_full = 'wasm64-emscripten'
+    #arch_full = 'wasm64-linux'
+    toolchain_names = ['emsdk']
+    compatible_hosts = ['fedora', 'debian']
+    exe_wrapper_def = ""
+
+    def get_cross_config(self):
+        return {
+            'binaries': self.binaries,
+            'exe_wrapper_def': '',
+            'root_path': self.root_path,
+            'extra_libs': [],
+            'extra_cflags': [],
+            'host_machine': {
+                'system': 'emscripten',
+                'lsystem': 'emscripten',
+                'cpu_family': 'wasm64',
+                'cpu': 'wasm64',
+                'endian': 'little',
+                'abi': ''
+            }
+        }
+
+    @property
+    def wasm_sdk(self):
+        return get_target_step('emsdk', self.name)
+
+    @property
+    def install_path(self):
+        return self.wasm_sdk.install_path
+
+    @property
+    def root_path(self):
+        return self.install_path
+
+    @property
+    def binaries(self):
+        binaries = (('CC', 'emcc'),
+                    ('CXX', 'em++'),
+                    ('AR', 'emar'),
+                    ('STRIP', 'emstrip'),
+                    ('WINDRES', 'windres'),
+                    ('RANLIB', 'emranlib'),
+                    ('LD', 'wasm-ld'))
+        binaries = {k:pj(self.install_path, 'upstream', 'emscripten', v)
+                    for k,v in binaries}
+        binaries['PKGCONFIG'] = 'pkg-config'
+        return binaries
+
+    @property
+    def configure_option(self):
+        #return ""
+        return '--host={}'.format(self.arch_full)
+
+    @property
+    def configure_prefix(self):
+        return "emconfigure"
+
+    @property
+    def make_prefix(self):
+        return "emmake"
```

Comment:
I don't fully understand how these prefixes are supposed to work, but my impression was that both prefixes should be simply "em".

Reply:
In fact they are not prefix but wrapper (and I will rename them)

They are tools provided by emsdk to run configure. Something like `emconfigure source/configure --options` where we would simply run `source/configure --options` in other case.

They "simply" set the environment before running the real command.